server/csharp/TicketHub/Controllers/TicketController.cs

```diff
@@ -204,13 +206,48 @@ public struct PolicyResult
         public UCASTNode? Conditions;
     }
 
-    private async Task<UCASTNode?> getConditions(HttpContext context, string path, object input)
+    public struct CompileUCASTResult
     {
-        var authzService = context.RequestServices.GetRequiredService<OpaAuthzService>();
-        OpaClient opa = authzService.GetClient();
+        [JsonProperty("result")]
+        public CompileResult Result;
 
-        PolicyResult result = await opa.evaluate<PolicyResult>(path, input);
-        _logger.LogInformation(JsonConvert.SerializeObject(result));
-        return result.Conditions;
+        public struct CompileResult
+        {
+            [JsonProperty("query", NullValueHandling = NullValueHandling.Ignore)]
+            public UCASTNode? Conditions;
+        }
+    }
+
+    private async Task<UCASTNode?> getConditions(HttpContext context, string query, object input)
+    {
+        var url = opaURL + "/v1/compile";
+        using var client = new HttpClient();

```

Comment:
I learned about the [`using` keyword](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/using) today, courtesy of an IDE warning. :slightly_smiling_face: 

It ensures the resource is correctly disposed of, even if exceptions occur. When not followed by curly braces indicating a scope-of-usage, it just assumes the scope is the entire rest of the method, which is convenient. I originally had a scope indicator, but it was possible to simplify to this more concise form.